src/lib/git/GitRequest.ts

```diff
@@ -4,13 +4,16 @@
 class GitRequest {
```

Comment:
Not sure if I like the separation between `GitBackend` and `GitFrontend`. I think of git as a core data structure. Any interface or communication should be happening outside of Git. We will use git only to be managing history and packaging up pack files. The grpc code should be occurring outside of `src/lib/git`. 

Actually now that I think of it. You can just use `./src/...` without separating into `./src/lib/`. Just keep a directory `./src/bin` for the CLI top level programs:

```
./src/bin/pk
./src/bin/pk-agent
```

Everything else can be raised to `./src/*`.

Reply:
Git is not only acting as a data structure and way to create packfiles, but also as a protocol for how to send those packfiles over network requests (e.g. sideband protocols). So I can't really see a way to separate git from the network code.

Reply:
what we can do is separate the gRPC from the git stuff, but what then to call the gRPC stuff? it will still be only related to git because the gitService is purely for git and only git

Reply:
I have raised everything up to just `./src/*`

Reply:
yeah we are doing git over gRPC atm, its purely a transport for git at this stage.